volumes/miovision/sql/README.md

```diff
@@ -1,3 +1,549 @@
+
+## Identifying Questionable Data Quality
+
+The table `miovision_api.anomalous_ranges` is used to flag times and places in the data where counts are potentially questionable, suspicious, or irregular. Sometimes counts are clearly not valid or data are missing, and sometimes counts look very weird but have been investigated and confirmed as valid or subject to some specific caveat. This table helps us track the status of these investigations and their results. 
+The basic idea is to identify sections of data (by timerange, intersection, and classification) that have been flagged as suspicious/anomalous for whatever reason, and state what we know about that data-subset in a semi-structured way. This lets us further investigate where that is required and carve out data that has been formally cast into doubt from queries that need only-the-best.
+
+### An applied example
+
+When looking for only "typical" data, `anomalous_ranges` should be used along with tables like `ref.holiday` to filter data. 
+
+```sql
+SELECT volume_uid
+FROM miovision_api.volumes
+WHERE 
+    NOT EXISTS ( -- this is our one big filter for bad/dubious data
+        SELECT 1
+        FROM miovision_api.anomalous_ranges
+        WHERE
+            anomalous_ranges.problem_level IN ('do-not-use', 'questionable')
+            AND (
+                volumes.datetime_bin >= anomalous_ranges.range_start
+                OR anomalous_ranges.range_start IS NULL
+            ) AND (
+                volumes.datetime_bin < anomalous_ranges.range_end
+                OR anomalous_ranges.range_end IS NULL
+            ) AND (
+                anomalous_ranges.intersection_uid = volumes.intersection_uid
+                OR anomalous_ranges.intersection_uid IS NULL
+            ) AND (
+                anomalous_ranges.classification_uid = volumes.classification_uid
+                OR anomalous_ranges.classification_uid IS NULL
+            )
+    )
+    AND NOT EXISTS ( -- also exclude official holidays
+        SELECT 1 FROM ref.holiday WHERE holiday.dt = volumes.datetime_bin::date
+    )
+    AND etc.
+```
+
+### Identifying new anomalies
+The `anomalous_ranges` table is populated in different ways:
+  - manually by `miovision_data_detectives` after visual inspection or prompting from an Airflow alert
+  - automatically by a [daily script](function/function-identify-zero-counts.sql) which identifys zero volume days by intersection/classification.
+    - There is an intention to eventually flag more unusual volumes automagically (see [Issue #630](https://github.com/CityofToronto/bdit_data-sources/issues/630)). 
```

Comment:
can update this as well

Reply:
This comment was referring to other auto-flagging ideas brought up in that issue!